lib/cc/config/yaml/validator.rb

```diff
@@ -48,10 +48,19 @@ def invalid_engine_warnings
 
         def invalid_engines
           return [] unless cc_yaml_processable?
-          config = CC::Config::YAML.new(path)
-          config.engines.reject do |engine|
-            engine_exists?(engine)
-          end
+          config = CC::Config::YAMLAdapter.load(path)
+          config.config["plugins"]
+            .map do |name, data|
+              Engine.new(
+                name,
+                enabled: data.fetch("enabled", true),
+                channel: data["channel"],
+                config: data["config"],
+              )
+            end.
+            reject do |engine|
```

Comment:
Should we error in the case of an unknown engine instead of silently pruning the entry?

Reply:
We're pruning entries that *do* exist. The validator constructs all engines, deletes ones that are valid, and reports on the remaining ones.

I am planning to rebuild the validator in depth after stabilizing the rest of this: the current changes here were to keep specs passing/behavior consistent in the short term.